src/build_test.cc

```diff
@@ -602,19 +648,33 @@ bool FakeCommandRunner::WaitForCommand(Result* result) {
     result->status = ExitFailure;
   else
     result->status = ExitSuccess;
-  last_command_ = NULL;
+
+  // Provide a way for test cases to verify when an edge finishes that
+  // some other edge is still active.  This is useful for test cases
+  // covering behavior involving multiple active edges.
+  string const& verify_active_edge = edge->GetBinding("verify_active_edge");
+  if (!verify_active_edge.empty()) {
+    bool verify_active_edge_found = false;
+    for (vector<Edge*>::iterator i = active_edges_.begin(),
+         e = active_edges_.end(); !verify_active_edge_found && i != e; ++i) {
```

Comment:
I would remove `!verify_active_edge_found && ` and use break below (maybe even skip that as this is only used in one test and not performance critical). While creating `e` is faster than calling `end()` in the condition, I wouldn't use that trick here as it makes the loop more complicated as it needs to be.